sdk/helper/pluginutil/run_config.go

```diff
@@ -0,0 +1,154 @@
+package pluginutil
+
+import (
+	"context"
+	"crypto/sha256"
+	"crypto/tls"
+	"fmt"
+	"os/exec"
+
+	log "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-plugin"
+	"github.com/hashicorp/vault/sdk/version"
+)
+
+type runConfig struct {
+	// Provided by PluginRunner
+	command string
+	args    []string
+	sha256  []byte
+
+	// Initialized with what's in PluginRunner.Env, but can be added to
+	env []string
+
+	wrapper        RunnerUtil
+	pluginSets     map[int]plugin.PluginSet
+	hs             plugin.HandshakeConfig
+	logger         log.Logger
+	isMetadataMode bool
+	autoMTLS       bool
+}
+
+func (rc runConfig) run(ctx context.Context) (*plugin.Client, error) {
+	cmd := exec.Command(rc.command, rc.args...)
+	cmd.Env = append(cmd.Env, rc.env...)
+
+	// Add the mlock setting to the ENV of the plugin
+	if rc.wrapper != nil && rc.wrapper.MlockEnabled() {
+		cmd.Env = append(cmd.Env, fmt.Sprintf("%s=%s", PluginMlockEnabled, "true"))
+	}
+	cmd.Env = append(cmd.Env, fmt.Sprintf("%s=%s", PluginVaultVersionEnv, version.GetVersion().Version))
+
+	var clientTLSConfig *tls.Config
+	if !rc.isMetadataMode {
+		// Add the metadata mode ENV and set it to false
+		cmd.Env = append(cmd.Env, fmt.Sprintf("%s=%s", PluginMetadataModeEnv, "false"))
+
+		// Get a CA TLS Certificate
+		certBytes, key, err := generateCert()
```

Comment:
I think we want to avoid creating these certs and then response wrapping them if we are using AutoMTLS, right?

Reply:
It doesn't seem to make a difference in terms of using the plugin. I can certainly make this conditional but I defer that decision that knows go-plugin better than I do.

For what it's worth, this is a copy-paste of the previous `runCommon` function, so nothing is changed here other than `AutoMTLS` and how it gets its values.

Reply:
https://github.com/hashicorp/go-plugin/blob/4c2f0639bfb69d99f6f21b350e87639b4df32df2/client.go#L184-L205

> The client should not set TLSConfig, nor should the server set a TLSProvider, because AutoMTLS implies that a new certificate and tls configuration will be generated at startup.

Reply:
Done. I also included some tests around the ClientConfig creation.